src/scaffold.js

```diff
@@ -0,0 +1,193 @@
+const fs = require('fs-extra')
+const path = require('path')
+const prettier = require('prettier')
+const { getSubgraphBasename } = require('./command-helpers/subgraph')
+const { step } = require('./command-helpers/spinner')
+const { ascTypeForEthereum, valueTypeForAsc } = require('./codegen/types')
+
+const abiEvents = abi => abi.filter(item => item.type === 'event')
+
+// package.json
+
+const generatePackageJson = ({ subgraphName }) =>
+  prettier.format(
+    JSON.stringify({
+      name: getSubgraphBasename(subgraphName),
+      license: 'UNLICENSED',
+      scripts: {
+        codegen: 'graph codegen',
+        build: 'graph build',
+        deploy:
+          `graph deploy ` +
+          `--node https://api.thegraph.com/deploy/ ` +
+          `--ipfs https://api.thegraph.com/ipfs/ ` +
+          subgraphName,
+        'create-local': `graph create --node http://localhost:8020/ ${subgraphName}`,
+        'remove-local': `graph remove --node http://localhost:8020/ ${subgraphName}`,
+        'deploy-local':
+          `graph deploy ` +
+          `--node http://localhost:8020/ ` +
+          `--ipfs http://localhost:5001 ` +
+          subgraphName,
+      },
+      dependencies: {
+        '@graphprotocol/graph-cli': '0.7.2',
+        '@graphprotocol/graph-ts': '0.5.1',
+      },
+    }),
+    { parser: 'json' }
+  )
+
+// Subgraph manifest
+
+const eventSignature = event =>
+  `${event.name}(${event.inputs.map(input => input.type).join(',')})`
+
+const generateManifest = ({ abi, address, network, subgraphName }) =>
+  prettier.format(
+    `
+specVersion: 0.0.1
+schema:
+  file: ./schema.graphql
+dataSources:
+  - kind: ethereum/contract
+    name: Contract
+    network: ${network}
+    source:
+      address: '${address}'
+      abi: Contract
+    mapping:
+      kind: ethereum/events
+      apiVersion: 0.0.1
+      language: wasm/assemblyscript
+      entities:
+        ${abiEvents(abi)
+          .map(event => `- ${event.name}`)
+          .join('\n        ')}
+      abis:
+        - name: Contract
+          file: ./abis/Contract.json
+      eventHandlers:
+        ${abiEvents(abi)
+          .map(
+            event => `
+        - event: ${eventSignature(event)}
+          handler: handle${event.name}`
+          )
+          .join('')}
+      file: ./src/mapping.ts
+`,
+    { parser: 'yaml' }
+  )
+
+// Schema
+
+const ethereumTypeToGraphQL = name => {
+  let ascType = ascTypeForEthereum(name)
+  return valueTypeForAsc(ascType)
+}
+
+const generateEventType = event => `type ${event.name} @entity {
+      id: ID!
+      ${event.inputs
+        // TODO: Support inputs without names
```

Comment:
Out of curiosity, what sort of inputs have no name? How does this code behave in that case?

Reply:
Oops, that comment is outdated — we fall back to `param${index}` for inputs with no names now.

I don't recall where we encountered event inputs without names, but I'm certain we did somewhere. That's why the code generation does the same — if an input has no name, it uses `param${index}`.

Reply:
So the comment can be removed?

Reply:
Yes, I thought I had pushed a commit already. Done that now.